### PR TITLE
magento/magento2#10045 Rename item resolver to item provider

### PR DESCRIPTION
**Pull request by piotrkwiecinski**
### Description

Follow up PR to address naming as it'll be released for 2.3 there should be no BC here.

### Fixed Issues (if relevant)

1. magento/magento2#10045: Refactoring: Adding items to the Sitemap

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by piotrkwiecinski** [comment]
FYI @antonkril

**Comment by antonkril** [comment]
Thank you.

File: app/code/Magento/Sitemap/Model/ItemProvider/Composite.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function getItems($storeId)
     {
         $items = [];
 
-        foreach ($this->itemResolvers as $resolver) {
+        foreach ($this->itemProviders as $resolver) {
             foreach ($resolver->getItems($storeId) as $item) {
                 $items[] = $item;
```
Comment (Koc) [inline app/code/Magento/Sitemap/Model/ItemProvider/Composite.php:37]: Do we really collect all items in array? Why no use generators for lower memory usage?

Comment (piotrkwiecinski) [reply app/code/Magento/Sitemap/Model/ItemProvider/Composite.php:37]: @Koc goal of this pull request was to improve way to customise for 3rd party extensions to add how items are added without need to extend or override sitemap class itself. I haven't changed way how data is extracted from database itself and passed around. This is way I came up with while working on it in my free personal time and it was accepted as good enough solution. If you are able to propose more efficient way feel free to create and issue with your ideas as a way to start discussion or even better PR with an implementation.

**Pull request merged**